webpack.config.js

```diff
@@ -25,8 +25,13 @@ const webpackConfig = {
 	module: {
 		rules: [
 			{
-				test: /\.(t|j)sx?$/,
-				use: [ 'ts-loader', 'babel-loader' ],
+				test: /\.tsx?$/,
+				loader: 'ts-loader',
+				exclude: /node_modules/,
```

Comment:
I'm afraid there were a reason to use babel-loader for ts files. @reykjalin @ismaeldcom do you know/remember anything about it?

Reply:
I found a conflict between my change in `babel.config.json` and using babel for TypeScript files. It looks like babel's TypeScript preset has separate configuration which needs to be looked into.

Reply:
Yes, you can find some docs here https://babeljs.io/docs/en/babel-plugin-transform-typescript#typescript-compiler-options

Reply:
I am more into removing a typescript preset in favor of `ts-loader` as it will reduce number of configurations we have to keep an eye on.

Reply:
Done! Verified with #1892

Reply:
I followed the [React conversion guide](https://github.com/Microsoft/TypeScript-React-Conversion-Guide#typescript-react-conversion-guide) that's linked to from TypeScript's [migrate from JS tutorial](https://www.typescriptlang.org/docs/handbook/migrating-from-javascript.html).

What they say there makes me inclined to believe we should keep the `ts-loader` build step, which makes sure [webpack will use `tsc` to compile TypeScript](https://www.typescriptlang.org/docs/handbook/integrating-with-build-tools.html#webpack).

> replace the Babel build step with TypeScript, as TypeScript also offers transpiling to lower ECMAScript versions and JSX transpilation with a shorter build time in most cases.

If I recall correctly, the only reason I kept Babel in was to make sure our JS presets are respected, e.g. IE11 support. Ideally we'd compile the whole thing with TypeScript.

Reply:
I suppose the actionable item I'm trying to communicate here ^ is that _eventually_ we'll want to remove Babel entirely from our build pipeline. I don't think this is the PR to do that in, since I think migrating everything from JS to TS should be completed first 🙂 

Not sure if that helps you at all with dealing with this specific issue now though, sorry!